quickjs/src/main/jni/Context.cpp

```diff
@@ -754,6 +766,26 @@ JSValue Context::throwJavaExceptionFromJs() const {
   return JS_ThrowInternalError(jsContext, "Java Exception");
 }
 
+JNIEnv* Context::getEnv() const {
+  JNIEnv* env = nullptr;
+  javaVm->GetEnv(reinterpret_cast<void**>(&env), jniVersion);
+  if (env) {
+    return env;
+  }
+
+  javaVm->AttachCurrentThread(
+#ifdef __ANDROID__
+      &env,
+#else
+      reinterpret_cast<void**>(&env),
+#endif
+      nullptr);
+  if (env) {
+    thread_local jniEnvDetacher detacher(javaVm);
```

Comment:
Is this cleared if the thread goes away? This is the only part I don't think I understand.

Reply:
Yeah, `thread_local` basically keeps the object alive until the thread exit, and then the destructor for `jniEnvDetacher` will get called and disconnect it from the JVM.